demo/xid-error-mitigation.yaml

```diff
@@ -0,0 +1,140 @@
+## Opt-in Error Codes Detection in Device Plugin Health Checker
+
+### Why do we want to make it opt-in rather than hard-coded in device plugin?
+
+If it's a hardware issue, we want to catch that in the health check and kubectl
+describe nodes will show it to customers so that they can migrate their
+workloads. The next step is GKE migrate the workload through graceful node
+shutdown.
+
+xid 79 could be a software issue. Migrating the workload to other nodes could
+have the same issue. We only want health check code to catch hardware errors, so
+we won't hardcode 79 in the health check code. We want the customers to do their
+investigation to rule out the possible user errors before treating it as a
+hardware failure. Once the customers actively opt in with configuration files,
+we assume they are aware of the possible user errors and they intend to treat 79
+as a hardware failure.
+
+### Xid Code ConfigMap
+
+Create a yaml file like this and put the xid codes you intend to add under
+data.HealthCriticalXid with the following format. Apply the yaml file to your
+cluster.
+
+```
+apiVersion: v1
```

Comment:
Can we add these as actual files in a directory with a README.md file explaining how to deploy it? Then the user could just clone the repo and deploy it from their commandline.